src/main/java/org/jenkinsci/plugins/workflow/steps/StepContext.java

```diff
@@ -149,4 +149,9 @@ public boolean hasBody() {
     public abstract int hashCode();
 
     private static final long serialVersionUID = 1L;
+
+    /**
+     * Returns the function name used in pipeline Jenkinsfile for the current StepNode.
+     */
+    public abstract String getStepDisplayFunctionName() throws IOException;
```

Comment:
Adding such a method is one approach, but my suggestion was rather to let `DefaultStepContext` expose `StepDescriptor` from `get` as a derivative of `FlowNode`, consistent with the other objects it is documented you can obtain from `StepContext`, so that in this PR you would

```java
var desc = context.get(StepDescriptor.class);
String functionName;
if (desc != null) {
    functionName = desc.getFunctionName();
} else {
    functionName = "<unknown step>";
}
```

Reply:
sorry I didn't understand at beginning, got it working now. 
ty